sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -441,6 +441,20 @@ void gnrc_ipv6_nib_change_rtr_adv_iface(gnrc_netif_t *netif, bool enable)
     }
     gnrc_netif_release(netif);
 }
+
+void gnrc_ipv6_nib_change_rtr_adv_rio_iface(gnrc_netif_t *netif, bool enable)
+{
+    gnrc_netif_acquire(netif);
+    if (enable) {
+        netif->flags |= GNRC_NETIF_FLAGS_IPV6_RTR_ADV_RIO;
+    }
+    else {
+        netif->flags &= ~GNRC_NETIF_FLAGS_IPV6_RTR_ADV_RIO;
+    }
+    gnrc_netif_release(netif);
```

Comment:
The `netif` lock is a recursive lock, so it should be safe(r) to move the release after the rtr_adv enable.

Reply:
Ok